### PR TITLE
 migrate babe and authorship to use derive-impl

### PR DESCRIPTION
**Pull request by Dmitry-Borodin**
Moving a babe and authorship pallets to the latest and greatest derive_impl.

Part of https://github.com/paritytech/polkadot-sdk/issues/171

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3876585

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3876587

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3876586

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3877224

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3877225

File: substrate/frame/authorship/src/lib.rs [hunk 1]
```diff
@@ -36,10 +35,11 @@ pub trait EventHandler<Author, BlockNumber> {
 
 #[frame_support::pallet]
 pub mod pallet {
-	use super::*;
 	use frame_support::pallet_prelude::*;
 	use frame_system::pallet_prelude::*;
 
+	use super::*;
+
```
Comment (liamaharon) [inline substrate/frame/authorship/src/lib.rs:42]: ?

**Review by franciscoaguirre — APPROVED** [review]

File: substrate/frame/authorship/src/lib.rs (hunk 1 above)
Comment (juangirini) [reply substrate/frame/authorship/src/lib.rs:42]: This has just been moved down, as pointed here https://github.com/paritytech/polkadot-sdk/pull/1790#discussion_r1345831064 this reformatting needs to be undone

File: substrate/frame/authorship/src/lib.rs [hunk 2]
```diff
@@ -22,9 +22,8 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use frame_support::traits::FindAuthor;
-use sp_std::prelude::*;
-
 pub use pallet::*;
+use sp_std::prelude::*;
```
[the remark this answers is not in the document]
Comment (Dmitry-Borodin) [reply substrate/frame/authorship/src/lib.rs]: Thx looks like intellij formatter was changing this, but rustfmt doesn't change back and acceping any order. Fixed everywhere.

**Review by KiChjang — APPROVED** [review]
If it compiles, it works

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3882857

File: substrate/frame/authorship/src/lib.rs (hunk 1 above)
Comment (Dmitry-Borodin) [reply substrate/frame/authorship/src/lib.rs]: All fixed. Thx for noticing.

**Review by sam0x17 — APPROVED** [review]

**Review by liamaharon — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3890770

**Review by juangirini — APPROVED** [review]

**Pull request merged**